v3/plugins/cdr/code-server/2.1523-vsc1.38.1-openshift/meta.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: v2
+publisher: cdr
+name: code-server
+version: 2.1523-vsc1.38.1-openshift
+type: Che Editor
+displayName: code-server
+title: Visual Studio Code - Web (code-server)
+description: A open source distribution of Visual Studio Code as a cloud IDE
+icon: https://raw.githubusercontent.com/sr229/discord-open-source/master/logos/code-server.svg 
+category: Editor
+repository: https://github.com/cdr/code-server
+firstPublicationDate: "2019-09-08"
+spec:
+  endpoints:
+   -  name: "code-server"
+      public: true
+      targetPort: 9000
+      attributes:
+        protocol: http
+        type: ide
+  containers:
```

Comment:
One OpenShift-specific security fix that is missing: containers cannot be created on some clusters if they mount a volume that isn't also mounted as a k8s/OpenShift PVC:
```
Error: Error response from daemon: create 84eb3536d3dc596a7f79926e3b1f922adc77a19e2c5eec2e0b038b7437e7dd58: error while creating volume path '/var/lib/docker/volumes/84eb3536d3dc596a7f79926e3b1f922adc77a19e2c5eec2e0b038b7437e7dd58/_data': mkdir /var/lib/docker/volumes/84eb3536d3dc596a7f79926e3b1f922adc77a19e2c5eec2e0b038b7437e7dd58: permission denied
```

To resolve this, you would need to mount any volumes from the dockerimage as PVCs. The good news is we recently added support for this -- you can specify
```
containers:
    volumes:           
        - mountPath:
          name:
          ephemeral: # boolean; if true volume will be ephemeral, otherwise volume will be persisted
```
Looking at your current Dockerfile, you would need to add mounts for `/home/coder/projects` and `/home/coder/certs` here.

Alternatively, since the openshift-specific image is built using its own dockerfile, I would consider removing the `VOLUME` directives, since you can always add volumes in your k8s/OpenShift yaml.

Reply:
Will do that right now, thanks for the feedback.